fta/samples/api/views.py

```diff
@@ -56,16 +56,40 @@ def add_fathom_sample(self, request, format=None):
         freeze_software = request.data["freeze_software"]
         notes = request.data["notes"] if "notes" in request.data else ""
         sample = sample_from_required(frozen_page, freeze_software, notes)
-        sample.save()
+        existing_sample = Sample.objects.filter(url__contains=sample.url).order_by(
+            "-id"
+        )[:1]
+
+        # If the sample already exists reuse it.
+        if existing_sample.exists() is False:
+            sample.save()
+        else:
+            sample = existing_sample.first()
+
+        # check if a LabeledSample already exists for the sample.
+        existing_labeled_sample = LabeledSample.objects.filter(
+            original_sample_id=sample.id
+        )
 
-        fta_sample, labels_to_fta_ids = convert_fathom_sample_to_labeled_sample(
+        id_of_existing_labeled_sample = -1
+        if existing_labeled_sample.exists() is True:
+            id_of_existing_labeled_sample = existing_labeled_sample.first().id
+
+        fta_sample, fta_ids_to_label = convert_fathom_sample_to_labeled_sample(
             frozen_page
         )
         labeled_sample, _ = LabeledSample.objects.get_or_create(
             original_sample=sample, modified_sample=fta_sample
         )
 
-        for fathom_label, fta_id in labels_to_fta_ids.items():
+        # If a LabeledSample already existed need to update the superseded_by field to the newly create LabeledSample.
+        if id_of_existing_labeled_sample != -1:
+            LabeledSample.objects.filter(id=id_of_existing_labeled_sample).update(
+                superseded_by=labeled_sample.id
+            )
```

Comment:
I don't think you need to explicitly get the `id` above and then filter the queryset again here for this. You should be able to just do:
```suggestion
        if existing_labeled_sample.exists():
            existing_labeled_sample.update(superseded_by=labeled_sample)
```

Reply:
I previously tried your suggested approach however the `existing_labeled_sample` QuerySet is lazy loaded.  Since I need to create the new `LabeledSample` to get the id to assign to the existing `LabeledSample` the result is the following (the new LabeledSample matches the query clause causing it to get updated as well).  Once I address your other suggestions this may be a moot issue.

```
>  select id, original_sample_id, superseded_by_id from samples_labeledsample;
>  id  | original_sample_id | superseded_by_id
> -----+--------------------+------------------
>  544 |                543 |              544
```

Reply:
Aah, tricky. Makes sense. I think you're right that with the new behavior this will be moot.

Reply:
I ultimately used the labeled_sample_id from the post.